src/postgres/commands/configurePostgresFirewall.ts

```diff
@@ -7,43 +7,74 @@ import PostgreSQLManagementClient from "azure-arm-postgresql";
 import { FirewallRule } from "azure-arm-postgresql/lib/models";
 import * as publicIp from 'public-ip';
 import * as vscode from 'vscode';
-import { createAzureClient, DialogResponses, IActionContext } from "vscode-azureextensionui";
+import { createAzureClient, DialogResponses, IActionContext, IAzureQuickPickItem } from "vscode-azureextensionui";
 import { ext } from "../../extensionVariables";
+import { IPostgresWizardContext } from "../../tree/PostgresAccountWizard/IPostgresWizardContext";
 import { azureUtils } from "../../utils/azureUtils";
 import { localize } from "../../utils/localize";
 import { nonNullProp } from "../../utils/nonNull";
 import { PostgresServerTreeItem } from "../tree/PostgresServerTreeItem";
 
-export async function configurePostgresFirewall(context: IActionContext, treeItem?: PostgresServerTreeItem): Promise<void> {
-    if (!treeItem) {
+export async function configurePostgresFirewall(context: IActionContext | IPostgresWizardContext, treeItem?: PostgresServerTreeItem, createMode?: boolean): Promise<string> {
+    const ip: string = await publicIp.v4();
+    if (!treeItem && createMode) {
+
+        const yesPick: IAzureQuickPickItem<string> = { data: 'Yes', label: 'Yes' };
+        const noPick: IAzureQuickPickItem<string> = { data: 'No', label: 'No' };
+        const picks: IAzureQuickPickItem<string>[] = [yesPick, noPick];
+
+        const wizardContext: IPostgresWizardContext = <IPostgresWizardContext>context;
+
+        const result = await ext.ui.showQuickPick(picks, { placeHolder: `Would you like to add firewall rule for your IP ${ip} to this server?` });
+
+        if (result.label === "Yes") {
+            wizardContext.firewall = true;
+        } else {
+            wizardContext.firewall = false;
+        }
+
+        return result.label;
+
+    } else if (!treeItem) {
         treeItem = <PostgresServerTreeItem>await ext.tree.showTreeItemPicker(PostgresServerTreeItem.contextValue, context);
     }
 
-    const ip: string = await publicIp.v4();
     await ext.ui.showWarningMessage(
         localize('firewallRuleWillBeAdded', 'A firewall rule for your IP ({0}) will be added to server "{1}". Would you like to continue?', ip, treeItem.server.name),
         { modal: true },
         { title: DialogResponses.yes.title }
     );
 
+    if (treeItem) {
+        void setFirewallRule(treeItem);
+    }
+
+    return "";
+}
+
+export async function setFirewallRule(treeItem: PostgresServerTreeItem): Promise<void> {
+
+    const ip: string = await publicIp.v4();
```

Comment:
I believe you already have `ip` and can pass it in to this function instead of calling `publicIp.v4()` again, which may or may not add a delay